### PR TITLE
feat/get all balances

### LINKED ISSUE #223: [SDK] Get all balances
Continuation of this task https://docs.google.com/document/d/1eM_tP0Ct_DjhSTf9Rdq5_r8fwj0JfRhiSNCknab495Y/edit

Expose a method on the sdk called `getAllBalances({ chainIds?: string[], cosmosAddress?: string, evmAddress?: string })`
- cosmosAddress will need to have its own type, according to how you built `getAllCosmosBalances` not a string

Which internally decides based on the argument passed if it's going to call to `getAlllEvmBalances` or `getAllCosmosBalances`

The main idea of `chainIds` argument is to allow only fetch balances given a chain, like by a single RPC I can get all token balances on ethereum
- if `chainIds` is provided, then only get balances for the chainIds provided.
- if `chainIds` is not provided, get balance for all chains compatible with the addresses given

### PR DESCRIPTION
**Pull request by genaroibc**
# Description

 closes #223 

Add methods to get EVM and Cosmos token balances.

```js
const squid = new Squid({
  baseUrl: "https://v2.api.squidrouter.com",
  integratorId: "squid-test"
});

await squid.init();

const cosmos = await squid.getAllCosmosBalances({
  addresses: [
    {
      chainId: "cosmoshub",
      address: "cosmos1awrua7e2kj69d7vn5qt5tccrhavmj9xajl58pw",
      coinType: 118
    }
  ]
});

const evmBalances = await squid.getAllEvmBalances({
  userAddress: "0x9a32549df3ff3c1fd725f81093c47445218de723",
  chains: [1, 43114, 8453]
});
```

### `getAllBalances`
```js
const balances = await squid.getAllBalances({
    chainIds: ["1", "43114", "osmosis-1"],
    evmAddress: "0x344b63c2BcB4B61765083735e8F49Bb203415a33",
    cosmosAddresses: [
      {
        address: "cosmos1awrua7e2kj69d7vn5qt5tccrhavmj9xajl58pw",
        chainId: "cosmoshub-4",
        coinType: 118
      }
    ]
  });

// result
{
  cosmos: [
    {
      amount: "676727870",
      denom: "ibc/E6931F78057F7CC5DA0FD6CEF82FF39373A6E0452BF1FD76910B93292CF356C1",
      chainId: "osmosis-1"
    }
  ],
  evm: [
    {
      symbol: "USDC",
      address: "0xB97EF9Ef8734C71904D8002F8b6Bc66Dd9c48a6E",
      decimals: 6,
      balance: "1549424225"
    }, {
      symbol: "AVAX",
      address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
      decimals: 18,
      balance: "41345097216001030"
    }, {
      symbol: "ETH",
      address: "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
      decimals: 18,
      balance: "33653125888601"
    }
  ]
}
```

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -503,3 +503,15 @@ export type CoinTypeAddress = {
 
 export const IBC_TRANSFER_TYPE = "/ibc.applications.transfer.v1.MsgTransfer";
 export const WASM_TYPE = "/cosmwasm.wasm.v1.MsgExecuteContract";
+
+export type TokenBalance = {
+  symbol: string;
+  address: string;
+  decimals: number;
+  balanceInDecimal: string;
```
Comment (jmdev3) [inline src/types/index.ts:511]: Rename this variable to balance, and store it in wei not in decimals, the format must be done outside on the implementation

File: src/index.ts [hunk 2]
```diff
@@ -730,6 +733,19 @@ export class Squid {
 
     return response.data.price;
   }
+
+  public async getAllEvmBalances({
```
Comment (jmdev3) [inline src/index.ts:737]: I would rename this function to something like `getAllTokenBalances`, and this must receive 2 addresses, one is evm and the other one is cosmos

Comment (ecdsafu) [reply src/index.ts:737]: For Cosmos remember we have multiple address types. So also need to support these.

**Review by odcey — CHANGES_REQUESTED** [review]
I would unify the response
- adding decimals on cosmos item
- naming balance instead of amount for cosmos to be the same as evm

**Review by jmdev3 — APPROVED** [review]
LGTM! Good work

**Comment by ecdsafu** [comment]
Exciting to have this ready, thanks a lot @genaroibc !

Could you please update the PR comments to reflect the updated interface?

**Comment by genaroibc** [comment]
Updated!

**Pull request merged**